--- app/js/app.js
+++ app/js/app.js
@@ -93,7 +93,7 @@
                   : orsApikeyFactory.getApiKey();
             let ak = "?api_key=" + apiKey;
             if (
-              config.url === ENV.geocode + "/autocomplete" ||
+              config.url === ENV.geocode + "/api" ||
               config.url === ENV.geocode + "/reverse" ||
               config.url.startsWith(ENV.directions) ||
               config.url === ENV.isochrones ||
