--- a/Gruntfile.default.js
+++ b/Gruntfile.default.js
@@ -192,7 +192,7 @@ module.exports = function(grunt) {
         connect: {
             dev: {
                 options: {
-                    hostname: "localhost",
+                    hostname: "0.0.0.0",
                     port: 3005,
                     //base: "src",
                     livereload: true,
@@ -213,7 +213,8 @@ module.exports = function(grunt) {
                                 "/node_modules",
                                 serveStatic("./node_modules")
                             ),
-                            proxy("/ors",{ target: 'http://localhost:8082', changeOrigin: true, pathRewrite: {'^/ors/': '/' }}),
+                            proxy("/ors/geocode/",{ target: 'XXX_GEOCODE_URL_BASE', changeOrigin: true}),
+                            proxy("/ors",{ target: 'http://openrouteservice-app:8080', changeOrigin: true}),
                             serveStatic("./app")
                         ];
                     }
@@ -269,19 +270,23 @@ module.exports = function(grunt) {
                 constants: {
                     ENV: {
                         name: "local",
-                        geocode:
-                            "http://localhost:3005/ors/geocode",
+                        geocode: "/ors/geocode",
                         directions:
-                            "http://localhost:3005/ors/v2/directions",
+                            "/ors/v2/directions",
                         isochrones:
-                            "http://localhost:3005/ors/isochrones",
-                        matrix:
-                            "http://localhost:3005/ors/matrix",
-                        pois:
-                            "https://api.openrouteservice.org/pois",
+                            "/ors/isochrones",
+                        matrix: "/ors/matrix",
+                        pois: "/ors/pois",
                         shortenlink: "https://api-ssl.bitly.com/v3/shorten",
                         mapsurfer:
                             "https://api.openrouteservice.org/mapsurfer/{z}/{x}/{y}.png",
+			tileLayers: {
+                            bkgGrey: "XXX_TILE_URL_BKG/{z}/{x}/{y}.png",
+                            osmDE: "XXX_TILE_URL_OSMDE/{z}/{x}/{y}.png",
+                            osmBasicPastHill: "XXX_TILE_URL_BASIC/{z}/{x}/{y}.png",
+                            labelsOverlay: "XXX_TILE_URL_LABELS/{z}/{x}/{y}.png"
+                        },
+                        osmAndTerrainAttribution: 'Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, terrain <a href="">ASTER</a> (a product of METI and NASA), <a href="https://www.earthenv.org/DEM">EarthEnv-DEM90</a> and <a href="https://open.canada.ca/data/en/dataset/7f245e4d-76c2-4caa-951a-45d1d2051333">Canadian Digital Elevation Model (contains information licensed under the Open Government Licence Canada)',
                         landmarks: "https://landmarks-api.openrouteservice.org/",
                         fuel: "https://api.openrouteservice.org/fuel"
                     }
@@ -294,16 +299,23 @@ module.exports = function(grunt) {
                 constants: {
                     ENV: {
                         name: "production",
-                        geocode: "https://api.openrouteservice.org/geocode",
+                        geocode: "/ors/geocode",
                         directions:
-                            "https://api.openrouteservice.org/v2/directions",
+                            "/ors/v2/directions",
                         isochrones:
-                            "https://api.openrouteservice.org/isochrones",
-                        matrix: "https://api.openrouteservice.org/matrix",
-                        pois: "https://api.openrouteservice.org/pois",
+                            "/ors/isochrones",
+                        matrix: "/ors/matrix",
+                        pois: "/ors/pois",
                         shortenlink: "https://api-ssl.bitly.com/v3/shorten",
                         mapsurfer:
                             "https://api.openrouteservice.org/mapsurfer/{z}/{x}/{y}.png",
+			tileLayers: {
+                            bkgGrey: "XXX_TILE_URL_BKG/{z}/{x}/{y}.png",
+                            osmDE: "XXX_TILE_URL_OSMDE/{z}/{x}/{y}.png",
+                            osmBasicPastHill: "XXX_TILE_URL_BASIC/{z}/{x}/{y}.png",
+                            labelsOverlay: "XXX_TILE_URL_LABELS/{z}/{x}/{y}.png"
+                        },
+                        osmAndTerrainAttribution: 'Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, terrain <a href="">ASTER</a> (a product of METI and NASA), <a href="https://www.earthenv.org/DEM">EarthEnv-DEM90</a> and <a href="https://open.canada.ca/data/en/dataset/7f245e4d-76c2-4caa-951a-45d1d2051333">Canadian Digital Elevation Model (contains information licensed under the Open Government Licence Canada)',
                         landmarks: "https://landmarks-api.openrouteservice.org/",
                         fuel: "https://api.openrouteservice.org/fuel"
                     }
@@ -398,7 +410,7 @@ module.exports = function(grunt) {
             "clean:task_rm_build_unused",
             "stripDebug",
             "cacheBust",
-            "connect:build:keepalive"
+            "connect:build"
         ]
     );
     grunt.registerTask("dev", "Run local server for development purposes", [
